src/rgw/services/svc_mdlog.h

```diff
@@ -75,11 +76,11 @@ class RGWSI_MDLog : public RGWServiceInstance
 
   // traverse all the way back to the beginning of the period history, and
   // return a cursor to the first period in a fully attached history
-  RGWPeriodHistory::Cursor find_oldest_period(const DoutPrefixProvider *dpp, optional_yield y);
+  RGWPeriodHistory::Cursor find_oldest_period(const DoutPrefixProvider *dpp, optional_yield y, rgw::sal::ConfigStore* cfgstore);
```

Comment:
this class `RGWSI_MDLog` already has a `cfgstore` member variable, why does this member function take it too? same for init_oldest_log_period and pull_period below